k4FWCore/include/k4FWCore/Consumer.h

```diff
@@ -90,6 +91,13 @@ namespace k4FWCore {
 
       // ... instead, they must implement the following operator
       virtual void operator()(const In&...) const = 0;
+
+      auto inputLocations(int i) const {
```

Comment:
Since this function is supposed to be used by algorithm developers a docstring could be useful

Reply:
Docstring added for all the new location functions.